core/templates/thread_work_pool.h

```diff
@@ -117,19 +120,32 @@ class ThreadWorkPool {
 
 	void end_work() {
 		ERR_FAIL_COND(current_work == nullptr);
-		for (uint32_t i = 0; i < thread_count; i++) {
+		for (uint32_t i = 0; i < threads_working; i++) {
 			threads[i].completed.wait();
 			threads[i].work = nullptr;
 		}
 
+		threads_working = 0;
 		memdelete(current_work);
 		current_work = nullptr;
 	}
 
 	template <class C, class M, class U>
 	void do_work(uint32_t p_elements, C *p_instance, M p_method, U p_userdata) {
-		begin_work(p_elements, p_instance, p_method, p_userdata);
-		end_work();
+		switch (p_elements) {
+			case 0:
+				// Nothing to do, so do nothing
```

Comment:
Missing trailing comma in comment.
```suggestion
				// Nothing to do, so do nothing.
```